### PR TITLE
Add ability to use replied message's content as `.randomcase` argument.

### LINKED ISSUE #1074: Use replied message as argument to `.randomcase` when text isn't provided.
## Description

Allow `.randomcase` to work on replied messages whenever no text argument is provided.

## Reasoning

Pretty self-explanatory; means you can just reply to a message instead of having to manually copy-paste the content as the command argument.

## Proposed Implementation

A simple if statement checking whether a text argument was provided, and if not whether a message was replied to.

## Would you like to implement this yourself?

- [x] I'd like to implement this feature myself
- [ ] Anyone can implement this feature

### PR DESCRIPTION
**Pull request by TizzySaurus**
Closes #1074.

## Description

As per title.

Users can now reply to a message to its contents as the `text` argument, thanks to a call to `botcore.utils.commands.clean_text_or_reply`.

image
image

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

File: bot/exts/fun/fun.py [hunk 1]
```diff
@@ -60,13 +61,14 @@ async def roll(self, ctx: Context, num_rolls: int = 1) -> None:
             raise BadArgument(f"`{Client.prefix}roll` only supports between 1 and 6 rolls.")
 
     @commands.command(name="randomcase", aliases=("rcase", "randomcaps", "rcaps",))
-    async def randomcase_command(self, ctx: Context, *, text: clean_content(fix_channel_mentions=True)) -> None:
+    async def randomcase_command(self, ctx: Context, *, text: str | None) -> None:
         """Randomly converts the casing of a given `text`."""
```
Comment (wookie184) [inline bot/exts/fun/fun.py:65]: Could you update the docstring to say the argument can be text, a reply, or a message link?

**Review by wookie184 — APPROVED** [review]
I think it would be nice if this supported `randomcase`ing replies to embeds, given this command already supports that through message links, although I don't really mind.

File: bot/exts/fun/fun.py (hunk 1 above)
Comment (wookie184) [reply bot/exts/fun/fun.py:65]: was meant to be a request changes as would like this to be done ^

**Comment by TizzySaurus** [comment]
> I think it would be nice if this supported `randomcase`ing replies to embeds, given this command already supports that through message links, although I don't really mind.

This isn't really possible to implement without making the code very message, since `clean_text_or_reply` doesn't have this feature. Users will at least still have the option of doing via message links (and said message link could be in the replied message).

**Review by Xithrius — APPROVED** [review]

**Pull request merged**